test/gtest/ternary_tensor_ops.cpp

```diff
@@ -0,0 +1,452 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/tensor_ops.hpp>
+#include "gtest_common.hpp"
+
+#define MIO_OPS_DEBUG 0
+
+static std::vector<std::vector<int>> tensorALensArr = {{32, 16, 8, 4, 4}, // tensor A
+                                                       {16, 20, 16, 8},
+                                                       {20, 16, 8},
+                                                       {1, 16, 8},
+                                                       {16, 8},
+                                                       {8}};
+
+static std::vector<std::vector<int>> tensorBLensArr = {{32, 16, 8, 4, 4}, // tensor B
+                                                       {32, 16, 1, 1, 1},
+                                                       {1, 16, 8, 1, 1},
+                                                       {1, 1, 8, 4, 1},
+                                                       {16, 20, 16, 8},
+                                                       {16, 20, 16, 1},
+                                                       {16, 20, 1, 1},
+                                                       {16, 1, 1, 1},
+                                                       {1, 20, 16, 8},
+                                                       {1, 20, 16, 1},
+                                                       {1, 20, 1, 1},
+                                                       {1, 1, 16, 8},
+                                                       {1, 1, 1, 8},
+                                                       {20, 16, 8},
+                                                       {20, 16, 1},
+                                                       {1, 16, 8},
+                                                       {1, 16, 1},
+                                                       {20, 1, 1},
+                                                       {16, 8},
+                                                       {16, 1},
+                                                       {1, 8},
+                                                       {8},
+                                                       {1}};
+
+static std::vector<std::vector<int64_t>> offsetsArr = {
+    {0, 0, 0}, {64, 32, 16}, {32, 16, 32}, {32, 16, 32}};
+
+static std::vector<std::vector<float>> alphabetaArr = {{1, 1, 0}, {-1, 1, 1}, {1.0, 0.5, 0.3}};
+
+static std::vector<std::vector<int>> stridesArr = {{8 * 16 * 20 * 16, 8 * 16 * 20, 8 * 16, 8, 1}};
```

Comment:
There is no point of having `vector` of `vectors`. It can be just a vector. There is no point to mimic old inefficient approach.